server/setup.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/pkg/errors"
+)
+
+const (
+	defaultAdminUsername = "sysadmin"
+	defaultAdminPassword = "Sys@dmin123"
+	defaultAdminEmail    = "success+sysadmin@simulator.amazonses.com"
+)
+
+func (p *Plugin) setupInstallation(install *Installation) error {
+	client := model.NewAPIv4Client(fmt.Sprintf("https://%s", install.DNS))
+
+	err := p.waitForDNS(client)
+	if err != nil {
+		return errors.Wrap(err, "encountered an error waiting for installation DNS")
+	}
+
+	err = p.createAndLoginAdminUser(client)
+	if err != nil {
+		return errors.Wrap(err, "encountered an error creating installation admin account")
+	}
+
+	err = p.setupInstallationConfiguration(client, install)
+	if err != nil {
+		return errors.Wrap(err, "encountered an error configuring the installation")
+	}
+
+	return nil
+}
+
+func (p *Plugin) waitForDNS(client *model.Client4) error {
+	for i := 0; i < 60; i++ {
+		_, resp := client.GetPing()
+		if resp.StatusCode == http.StatusOK {
+			return nil
+		}
+		if resp.Error != nil {
+			p.API.LogDebug(resp.Error.Error())
+		}
+		time.Sleep(time.Second * 10)
```

Comment:
How long does this take on average, something like 4-6 minutes? 10 minutes seems like a good upper bound.